tutorial/Chapter 1.md

```diff
@@ -35,7 +35,63 @@ Running code on the Wii U can be achieved through several methods.
+
+### Step 1: Install the devkitPro package manager
+The devkitPPC compiler toolchain is part of a larger set of software programs called devkitPro, which provides toolchains for development of homebrew for platforms such as the Nintendo 3DS, Switch, GameCube and Wii, as well as others. As such, devkitPro has its own package manager, `dkp-pacman` which can be installed relatively easily, and can be used to configure the development environment to your liking.
+
+For people using macOS or a Debian-based GNU/Linux distribution (e.g. Ubuntu, Raspbian) `dkp-pacman` can be installed by downloading and running the installer from its [download page](https://github.com/devkitPro/pacman/releases/latest). For other GNU/Linux distributions (especially Arch), you may already have the standard `pacman` installed, or you may be able to install `pacman`, which can be configured to read from the devkitPro repositories, as per the instructions on the [devkitPro wiki](https://devkitpro.org/wiki/devkitPro_pacman).
+
+### Step 2: Install devkitPPC and other tools
+Now that `dkp-pacman` is installed (or you've configured `pacman` to search the devkitPro repositories) make sure that the list of packages is up-to-date.
+
+(Note: make sure that you use the correct program for your system - `dkp-pacman` or the standard `pacman`)
+
+```
+sudo dkp-pacman -Sy
+```
+
+At any point after the installation, you can upgrade all installed packages to the latest version with:
+
+```
+sudo dkp-pacman -Syu
+```
+
+Now, the packages that you want installed to perform Wii U homebrew development include `devkitPPC` (a compiler and binary utilities for generic PowerPC processors), `wiiload` (used to launch a program on your console remotely) and `devkitpro-pkgbuild-helpers` (helpful utilities for linking to software libraries).
+
+```
+sudo dkp-pacman -S devkitPPC wiiload devkitpro-pkgbuild-helpers
+```
+
+### Step 3: Set up Fling (Wii U-specific package repository)
+In addition to the official devkitPro software packages, we have our own package repository (fling.heyquark.com) which contains many packages specific to Wii U development.
+
+The process of setting up Fling is documented on its [GitLab repository page](https://gitlab.com/QuarkTheAwesome/wiiu-fling).
+
+### Step 4: Installing wut
+With Fling installed, installing wut is very easy.
+
+For GNU/Linux and Windows Subsystem for Linux users:
+
+```
+sudo dkp-pacman -S wut-linux-bin
+```
+
+And for macOS users:
+
+```
+sudo dkp-pacman -S wut-osx
+```
+
+In both cases, this will install wut into `/opt/wut`. The wut build system requires that the environment variable `WUT_ROOT` be set to this directory, so run:
+```
+export WUT_ROOT=/opt/wut
+```
+
+This is only temporary, so you may wish to add this to your `.bash_profile` or `.bashrc` or whatever script your shell program runs at the beginning of a session.
```

Comment:
`wut-linux` adds a file to /etc/profile that should do this automatically, though you're right in that neither `wut-linux-bin` or `wut-osx` do this. I could add it to `wut-linux-bin`, though honestly I have no idea how to approach it on OSX.

Reply:
quick note: devkitPro ships `devkit-env`, which does the same /etc/profile thing. IIRC it's not an explicit dependency of devkitPPC, so it should probably be added to the package list.

Reply:
`devkit-env` is not in my list of packages. I looked at the [repository](https://github.com/devkitPro/tool-packages) and it does seem to install a script into `/etc/profile.d/` which adds `DEVKITPRO` and `DEVKITPPC` variables.

`/etc/profile.d` does not exist on my Mac, hence why I don't see `devkit-env` in my local package list. I'll see if there's an alternative way to programmatically add variables to shell sessions.

Reply:
Okay, so according to https://devkitpro.org/wiki/devkitPro_pacman#Customising_Existing_Pacman_Install -

If you're using a pre-existing msys2 install or OSX then please also set environment variables as follows:
```shell
DEVKITPRO=/opt/devkitpro
DEVKITARM=/opt/devkitpro/devkitARM
DEVKITPPC=/opt/devkitpro/devkitPPC
```

So, I think `wut-linux` and `wut-linux-bin` should add a file to `/etc/profile.d` which adds `WUT_ROOT` to the `PATH` automatically, but on macOS, much like with devkitPPC itself, the variables should be added to the user's `.bash_profile` manually.

Reply:
It seems that msys2 also has an `/etc/profile.d`, so Windows users *should* be able to use devkit-env (though for whatever reason devkitPro doesn't ship it). If we ever get a `wut-windows` - should be soon, the endian is fixed - then I'll probably drop something in /etc/profile.d.
As I mentioned elsewhere, I'm not sure how long `wut-linux-bin` is sticking around, but if I end up updating it I'll be sure to add an `/etc/profile.d` entry.

Not 100% on how to handle OSX, it may help to just have os-specific instructions.
> On Linux, install `devkit-env`. On Windows or OSX, run `echo EXPORT WUT_ROOT=... > ~/.bashrc`

Y'know? It'll at least work until devkitPro figures something out, then we can just copy them ;D